### DISCUSSION TITLE
Stake Registration and Staking Credential Transactions

### DISCUSSION THREAD

**Discussion by bapshago** [Python-Cardano/pycardano#169] [Q&A] [answered]
When registering a stake address on-chain, can you just provide the stake verification hash or is a script needed when creating the stake credentials and registration?

Using the code in the sample.txt, I keep getting the following error from blockfrost:  When the transaction is ran without the "certificates" it works without issue.

"blockfrost.utils.ApiError: {'error': 'Bad Request', 'message': '"transaction read error RawCborDecodeError [DecoderErrorDeserialiseFailure \\"Byron Tx\\" (DeserialiseFailure 1 \\"Size mismatch when decoding TxAux.\\\\nExpected 2, but found 4.\\"),DecoderErrorDeserialiseFailure \\"Shelley Tx\\" (DeserialiseFailure 1 \\"Size mismatch when decoding \\\\nRecord RecD.\\\\nExpected 4, but found 3.\\"),DecoderErrorDeserialiseFailure \\"Shelley Tx\\" (DeserialiseFailure 1 \\"Size mismatch when decoding \\\\nRecord RecD.\\\\nExpected 4, but found 3.\\"),DecoderErrorDeserialiseFailure \\"Shelley Tx\\" (DeserialiseFailure 1 \\"Size mismatch when decoding \\\\nRecord RecD.\\\\nExpected 4, but found 3.\\"),DecoderErrorDeserialiseFailure \\"Shelley Tx\\" (DeserialiseFailure 128 \\"expected bytes\\"),DecoderErrorDeserialiseFailure \\"Shelley Tx\\" (DeserialiseFailure 128 \\"expected bytes\\")]"', 'status_code': 400}"

[sample.txt](https://github.com/Python-Cardano/pycardano/files/10853154/sample.txt)

**Comment by bapshago** [comment]
cardano cli uses the following format when creating a stake.cert file:

{
    "type": "CertificateShelley",
    "description": "Stake Address Registration Certificate",
    "cborHex": "82008200581c28b78836ced904e42f77a9fbbbf27e49d5d5f651acbda1c12057d877"
}

does pycardano do this in the certificate module?  I looked in source and could not find it...

**Comment by cffls** [comment, marked as answer]
You can follow this example to create a stake registration: https://github.com/Python-Cardano/pycardano/blob/main/integration-test/test/test_certificate.py#L41-L44

**Reply by bapshago** [reply]
Using the test case I received a 404 from blockfrost on the utxo function. 

"blockfrost.utils.ApiError: {'status_code': 404, 'error': 'Not Found', 'message': 'The requested component has not been found.'}"

 I think it is because the address was never used; however it looks like the test case allots for that scenario by funding the address from a "giver" address so not sure why it was erroring.  

However, I took the middle part (where the address registers and delegates to a pool and used that to complete the transaction successfuly.  I am slowly figuring out how to work with pycardano. 

Thanks so much for the responses and help.  I am working on adding some tools to our stakepool website that allow a user to delegate via a browser wallet.  I have the sign transaction functionality up and running for authentication for stakepool delegators and it is working well.  

Thanks again for your help and the time you spent building the module.
